native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/internal/AgentCommandLineProvider.java

```diff
@@ -66,13 +70,18 @@ public AgentCommandLineProvider() {
     @OutputDirectory
     public abstract DirectoryProperty getOutputDirectory();
 
+    @Input
+    @Optional
+    public abstract ListProperty<String> getAgentOptions();
+
     @Override
     public Iterable<String> asArguments() {
         if (getEnabled().get()) {
             File outputDir = getOutputDirectory().getAsFile().get();
+            List<String> agentOptions = new ArrayList<>(getAgentOptions().getOrElse(Collections.emptyList()));
+            agentOptions.add("config-output-dir=" + outputDir.getAbsolutePath());
```

Comment:
Since the plugin is responsible for configuring `config-output-dir`, I think an attempt to set that by the user should result in a build failure.

In in the Maven support, I added a note to the reference documentation pointing that out, and in the code I have the following check.

```java
if (value.contains("config-output-dir")) {
    throw new IllegalStateException("config-output-dir cannot be supplied as an agent option");
}
```

I think we should add the same sort of check to the Gradle plugin.